app/services/order_purchaser.rb

```diff
@@ -55,15 +55,39 @@ def order_detail_updater
   end
 
   def order_in_the_past!
-    unless order.can_backdate_order_details?
+    unless can_backdate_order_details?
       raise NUCore::PurchaseException.new(I18n.t("controllers.orders.purchase.future_dating_error"))
     end
 
     # update order detail statuses if you've changed it while acting as
     if acting_as? && order_status_id.present?
-      order.backdate_order_details!(user, order_status_id)
-    else
-      order.complete_past_reservations!
+      backdate_order_details!(user, order_status_id)
+    end
+    complete_past_reservations!
+  end
+
+  def can_backdate_order_details?
+    order.ordered_at <= Time.zone.now
+  end
+
+  def backdate_order_details!(update_by, order_status)
```

Comment:
I think I do like these methods better here rather than on `Order`. Do you agree?

Reply:
Yeah, I think it makes sense to keep this functionality all in one file, and it is nice to keep the `Order` model a bit skinnier.